examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp

```diff
@@ -180,12 +188,14 @@ void BdxOtaSender::Reset()
 {
     mFabricIndex.ClearValue();
     mNodeId.ClearValue();
-    mTransfer.Reset();
-    if (mExchangeCtx != nullptr)
+
+    // mExchangeCtx->Close() will crash if ExchangeMgr is nullptr. So check for nullptr before calling Close().
+    if (mExchangeCtx && mExchangeCtx->GetExchangeMgr())
```

Comment:
How do we get into a state where we have a broken mExchangeCtx here (without an exchange manager)?

Reply:
I had literally the same comment here :) https://github.com/project-chip/connectedhomeip/pull/16616#discussion_r834599022

Reply:
When OTA Provider receives the first `BlockQuery` then it will try to open a file and if file designator is invalid, we should [abort the transfer](https://github.com/project-chip/connectedhomeip/blob/48f77750fd89f6fc2ac24cff68ea8559f3f81746/examples/ota-provider-app/ota-provider-common/BdxOtaSender.cpp#L136-L142). Abort will send a `StatusReport` to OTA Requestor, here OTA Provider receives an `Internal Error` from BDX and then a call `Reset()` will take us to this state.

Reply:
Let me be a little clearer.

The only way `GetExchangeMgr()` can possibly return null is if `~ExchangeContext` has been called.  So if you are ending up in this state, you are calling methods on potentially-deallocated memory.  Maybe this check will work.  And maybe it will crash.

What _should_ be happening is that whatever codepath causes `~ExchangeContext` to be called ensures that `mExchangeCtx` is nulled out at that point.  What codepath is that?

> and then a call `Reset()` will take us to this state.

No, I mean how did we get into a state where we are pointing to a destroyed exchange context, not how do we end up in this method.  What destroyed the exchange context?

Reply:
@bzbarsky-apple I have addressed this in #18425. Sorry for the very very late turnaround. Your suspicion was correct, when it send the StatusReport, SendMessage internally closes the exchange context and we were not setting it to null resulting in a crash.